pkg/http/sse.go

```diff
@@ -0,0 +1,160 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/ThreeDotsLabs/watermill/pubsub/gochannel"
+	"github.com/go-chi/render"
+)
+
+type StreamAdapter interface {
+	// GetResponse returns the response to be sent back to client.
+	// Any errors that occur should be handled and written to `w`, returning false as `ok`.
+	GetResponse(w http.ResponseWriter, r *http.Request) (response interface{}, ok bool)
+	// Validate validates if the incoming message should be handled by this handler.
+	// Typically this involves checking some kind of model ID.
+	Validate(r *http.Request, msg *message.Message) (ok bool)
+}
+
+type HandleErrorFunc func(w http.ResponseWriter, r *http.Request, err error)
+
+type defaultErrorResponse struct {
+	Error string `json:"error"`
+}
+
+// DefaultErrorHandler writes JSON error response along with Internal Server Error code (500).
+func DefaultErrorHandler(w http.ResponseWriter, r *http.Request, err error) {
+	w.WriteHeader(500)
+	render.Respond(w, r, defaultErrorResponse{Error: err.Error()})
+}
+
+type SSERouter struct {
+	fanOut       *gochannel.FanOut
+	errorHandler HandleErrorFunc
+	logger       watermill.LoggerAdapter
+}
+
+func NewSSERouter(
```

Comment:
Maybe we could add some documentation about what to pass as `upstreamRouter` here. `upstreamSubscriber` is maybe a little bit more intuitive, but what is the router here for?
What arguments to pass to the constructor?

I figure that there will be an example for SSE in due time, but I feel that a godoc might help

Reply:
Yeah, will add it. 👍